Sources/SwiftGenKit/Parsers/Filter.swift

```diff
@@ -0,0 +1,45 @@
+//
+// SwiftGenKit
+// Copyright © 2019 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+import PathKit
+
+public struct Filter {
+  public enum Error: Swift.Error, CustomStringConvertible {
+    case invalidRegex(pattern: String)
+
+    public var description: String {
+      switch self {
+      case .invalidRegex(let pattern):
+        return "error: Unable to parse regular expression '\(pattern)'."
+      }
+    }
+  }
+
+  fileprivate let regex: NSRegularExpression
+
+  /// Creates a filter with the given pattern, must be a valid regular expression.
+  ///
+  /// - Parameter pattern: The regular expression provided by the user
+  public init(pattern: String) throws {
+    do {
+      regex = try NSRegularExpression(pattern: pattern, options: [])
+    } catch {
+      throw Error.invalidRegex(pattern: pattern)
+    }
+  }
+}
+
+extension Path {
+  /// Checks if this path matches the given filter (regular expression).
+  ///
+  /// - Parameter filter: The regular expression to match
+  /// - Returns: true if it matches
+  func matches(filter: Filter) -> Bool {
+    let range = NSRange(string.startIndex..<string.endIndex, in: string)
+    return filter.regex.firstMatch(in: string, options: [], range: range) != nil
```

Comment:
I think it would be less surprising if we matches only the `basename` of the Path, instead of matching on its full absolute path?

Also, should we anchor the RegEx (there's an `options: […]` for that)? I personally think we shouldn't, and let the user anchor it using `^` and `$` if they need, and that will make easier RegExes for matching extensions using `\.xib$`, but then we should specify that in the docs on the `--filter` param to be precise enough to our users.

Reply:
Yeah, keep it unanchored, so that users can go loco if they want.

We're also not setting anything about case (in)sensitivity. Some of our default filters do match case insensitively with the `i` flag: `stuff\.(?i:json)$` matches `stuff.json`, `stuff.Json`, ...